language/functional_tests/tests/testsuite/natives/vector/vector_reverse.mvir

```diff
@@ -0,0 +1,32 @@
+import 0x0.Vector;
+
+main() {
+  let vec: Vector.T<u64>;
+  let is_empty: bool;
+  vec = Vector.empty<u64>();
+  is_empty = Vector.is_empty<u64>(&vec);
+
+  // empty case
+  Vector.reverse<u64>(&mut vec);
+  assert(move(is_empty) == Vector.is_empty<u64>(&vec), 0);
+
+  // non-empty case
+  Vector.push_back<u64>(&mut vec, 0);
+  Vector.push_back<u64>(&mut vec, 1);
+  Vector.push_back<u64>(&mut vec, 2);
+  Vector.push_back<u64>(&mut vec, 3);
+
+  assert(*Vector.borrow<u64>(&vec, 0) == 0, 1);
+  assert(*Vector.borrow<u64>(&vec, 1) == 1, 2);
+  assert(*Vector.borrow<u64>(&vec, 2) == 2, 3);
+  assert(*Vector.borrow<u64>(&vec, 3) == 3, 4);
+
+  Vector.reverse<u64>(&mut vec);
+
+  assert(*Vector.borrow<u64>(&vec, 3) == 0, 5);
+  assert(*Vector.borrow<u64>(&vec, 2) == 1, 6);
+  assert(*Vector.borrow<u64>(&vec, 1) == 2, 7);
+  assert(*Vector.borrow<u64>(&vec, 0) == 3, 8);
+
+  return;
+}
```

Comment:
Could you add tests for
- the size 1 case
- a vector containing an odd number of elements > 1?